app/controllers/api/functions.php

```diff
@@ -1139,73 +1483,96 @@
                 ->dynamic($execution, Response::MODEL_EXECUTION);
         }
 
-        $vars = array_reduce($function->getAttribute('vars', []), function (array $carry, Document $var) {
+        $durationStart = \microtime(true);
+
+        $vars = [];
+
+        // Shared vars
+        $vars = \array_merge($vars, \array_reduce($dbForProject->find('variables', [
```

Comment:
We should attach this as a subquery to the project objects like we do with other sub-collections. This way every time we fetch the project object we get this data and leverage our document level cache for much faster operations. This endpoint is performance critical and we should be very careful with every database query we perform here.

Reply:
Multiple reasons why I dont like using subquery here:

- It would act as read-only subquery. Pretty hard to document
- We now need to do clearCachedDocument() every time we update vars. If we miss a spot, very hard to find this bug
- Every project would need new attribute that would be always empty. It's only purpose is to define subquery to other collection

Reply:
Used subQuery for shared vars